### ISSUE TITLE
Custom tooltipSymbol

### ISSUE THREAD

**Issue by felixhao28** [codesuki/react-d3-components#16]
Currently, tooltipSymbol is merely a `symbol`. Yet I want more, for example, I need focus lines:

![image](https://cloud.githubusercontent.com/assets/2417391/6821197/2b2417b6-d310-11e4-860f-62c5ad3074df.png)

I implemented this by introducing a new attribute "tooltipSymbol", which is a function that accepts data, x and y. The code for my example is (coffeescript):

```coffee
<LineChart
    tooltipSymbol={(data, x, y) ->
        <g className="util-focus">
            <line
                className="focusLine"
                x1={x}
                y1={yScale(yDomain[0])}
                x2={x}
                y2={yScale(yDomain[1])}
            />
            <line
                className="focusLine"
                x1={xScale(xDomain[0])}
                y1={y}
                x2={xScale(xDomain[1])}
                y2={y}
            />
            <path
                className="dot"
                d=d3.svg.symbol().type("circle")()
                transform="translate(#{x}, #{y})"
                fill="red"
            />
        </g>
    }
    ...
/>
```

This might be needed for other people as well. Since it gives an easy way to access the x and y of selected data point and appending additional stuff on the chart.

**Comment by serveace** [comment]
One thing I'd suggest is to have the ability to pass a react component to render the element.  It's much cleaner and we use it internally for my company's custom widgets.  See: react-widgets http://jquense.github.io/react-widgets/docs/#/combobox for an example

**Comment by codesuki** [comment]
Good idea! I was seeing those focus lines in some d3 examples and thought of enabling them somehow.
I like your idea of people providing their own elements, that makes the library a little bit lighter than providing.

Typed this comment last night and forgot to hit send :)
